test/e2e/gardener/shoot/create_update_delete.go

```diff
@@ -20,178 +20,245 @@ import (
+							}
 						}
-						expectedNodeLabels["worker.gardener.cloud/kubernetes-version"] = kubernetesVersion
+					}).Should(Succeed())
+				}, SpecTimeout(time.Minute))
 
-						nodeList := &corev1.NodeList{}
-						g.Expect(shootClient.Client().List(ctx, nodeList, client.MatchingLabels{
-							"worker.gardener.cloud/pool": workerPool.Name,
-						})).To(Succeed())
-						g.Expect(nodeList.Items).To(HaveLen(1), "worker pool %s should have exactly one Node", workerPool.Name)
+				It("Verify reported CIDRs", func(ctx SpecContext) {
+					// For workerless shoots, the status.networking section is not reported. Skip its verification accordingly.
+					Eventually(func(g Gomega) {
+						g.Expect(s.GardenClient.Get(ctx, client.ObjectKeyFromObject(s.Shoot), s.Shoot)).To(Succeed())
 
-						for key, value := range expectedNodeLabels {
-							g.Expect(nodeList.Items[0].Labels).To(HaveKeyWithValue(key, value), "worker pool %s should have expected labels", workerPool.Name)
+						networking := ptr.Deref(s.Shoot.Status.Networking, gardencorev1beta1.NetworkingStatus{})
+						if nodes := s.Shoot.Spec.Networking.Nodes; nodes != nil {
+							g.Expect(networking.Nodes).To(ConsistOf(*nodes))
+							g.Expect(networking.EgressCIDRs).To(ConsistOf(*nodes))
 						}
-					}
-				}).Should(Succeed())
+						if services := s.Shoot.Spec.Networking.Services; services != nil {
+							g.Expect(networking.Services).To(ConsistOf(*services))
+						}
+						if pods := s.Shoot.Spec.Networking.Pods; pods != nil {
+							g.Expect(networking.Pods).To(ConsistOf(*pods))
+						}
+					}).Should(Succeed())
+				}, SpecTimeout(time.Minute))
 
-				By("Verify reported CIDRs")
-				// For workerless shoots, the status.networking section is not reported. Skip its verification accordingly.
-				Eventually(func(g Gomega) {
-					g.Expect(f.GardenClient.Client().Get(ctx, client.ObjectKeyFromObject(f.Shoot), f.Shoot)).To(Succeed())
+				inclusterclient.VerifyInClusterAccessToAPIServer(s)
+			}
 
-					networking := ptr.Deref(f.Shoot.Status.Networking, gardencorev1beta1.NetworkingStatus{})
-					if nodes := f.Shoot.Spec.Networking.Nodes; nodes != nil {
-						g.Expect(networking.Nodes).To(ConsistOf(*nodes))
-						g.Expect(networking.EgressCIDRs).To(ConsistOf(*nodes))
-					}
-					if services := f.Shoot.Spec.Networking.Services; services != nil {
-						g.Expect(networking.Services).To(ConsistOf(*services))
-					}
-					if pods := f.Shoot.Spec.Networking.Pods; pods != nil {
-						g.Expect(networking.Pods).To(ConsistOf(*pods))
+			var (
+				zeroDowntimeJob *batchv1.Job
+
+				cloudProfile *gardencorev1beta1.CloudProfile
+
+				controlPlaneKubernetesVersion string
+				poolNameToKubernetesVersion   map[string]string
+			)
+
+			if v1beta1helper.IsHAControlPlaneConfigured(s.Shoot) {
+				It("Deploy zero-downtime validator job to ensure no gardener downtime while running subsequent operations", func(ctx SpecContext) {
```

Comment:
The job verifies the availability of the API server not gardener, right?
```suggestion
				It("Deploy zero-downtime validator job to ensure no API server downtime while running subsequent operations", func(ctx SpecContext) {
```